docs/sdkx-react-native/delegates.mdx

```diff
@@ -0,0 +1,227 @@
+
+### Helpshift session ended event {#session-delegates}
+
+This event gets fired when the Helpshift session ends.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionEnded') {
+        //Your code here.
+      }
+    });  
+```
+
+### Unread Message Count Event {#unreadMessageCount}
+
+This event gets fired when you call `RequestUnreadMessageCount(FetchFromServer)` API to request unread messages count.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        //Your code here.
+      } 
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+       //Your code here.
+    });
+```
+### Conversation Status Event {#conversation-status}
+
+This event contains information about the current ongoing conversation.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationStatus') {
+        //Your code here.
+        //Event data: `latestIssueId`, `latestIssuePublishId`, `open`
+      }
+    });  
+```
+### Widget Toggle Event {#widget-toggle-event}
+
+This event is triggered when the user opens or exits the chat screen. This event is triggered with a boolean value of `"visible"` key. For your reference, see the below example:
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'widgetToggle') {
+        //Your code here.
+      }
+    });  
+```
+### Conversation Start Event {#conversation-start-event}
+
+This event triggers when the user sends the first message in a conversation. The event data object has a key, `message`, which includes the message string the end-user sent to start the conversation. For your reference, see the below example.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationStart') {
+        //Your code here.
+      }
+    });  
+```
+### Message Add Event {#message-add-event}
+
+This event is triggered when the user adds a message to a conversation. It might be a text message, a response via bot input, or an attachment. The event data object has the `type` and `body` keys, which indicate the type and body of the message added by the user. For your reference, see the below example.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'messageAdd') {
+        //Your code here.
+        //Event data keys: 
+        //  -` type` or `attachment` or `text`
+        //  - `body`
+      }
+    });  
+```
+
+### CSAT Submit Event {#csat-submit-event}
+
+This event is triggered when the user submits a CSAT rating after the conversation ends. The event data object has `rating` and `additionalFeedback` keys, which indicates the (star) rating and the additional comments provided by the user with the CSAT form. For your reference, see the below example.
```

Comment:
same here, show in code sample how to read rating and additionalFeedback

Reply:
Updated